components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt

```diff
@@ -113,6 +120,56 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
     }
 }
 
+/**
+ * Shares content via [ACTION_SEND] intent.
+ *
+ * @param text the data to be shared [EXTRA_TEXT]
+ * @param subject of the intent [EXTRA_TEXT]
+ * @return true it is able to share false otherwise.
+ */
+fun Context.shareMedia(
+    filePath: String,
+    contentType: String?,
+    subject: String? = null,
+    message: String? = null
+): Boolean {
+    val contentUri = FileProvider.getUriForFile(
+        this,
+        "${applicationContext.packageName}.feature.downloads.fileprovider", // (packageName + FILE_PROVIDER_EXTENSION)
+        File(filePath)
+    )
+
+    val intent = Intent().apply {
+        action = ACTION_SEND
+        type = contentType ?: contentResolver.getType(contentUri)
+        flags = FLAG_ACTIVITY_NEW_DOCUMENT or FLAG_GRANT_READ_URI_PERMISSION
+        putExtra(EXTRA_STREAM, contentUri)
+        if (subject != null) {
+            putExtra(EXTRA_SUBJECT, subject)
+        }
+        if (message != null) {
+            putExtra(EXTRA_TEXT, message)
+        }
+    }
+
+    val shareIntent = Intent.createChooser(intent, getString(R.string.mozac_support_ktx_menu_share_with)).apply {
```

Comment:
Saw a SecurityException posted on Android 10 & Android 11 when using `Intent.createChooser` with a FileProvider URI.
Same is happening for Chrome or WhatsApp when sharing files.
Created a new issue on Google's issue tracker - https://issuetracker.google.com/issues/177734533